AutoRest/Generators/Ruby/Azure.Ruby.Tests/RspecTests/lro_spec.rb

```diff
@@ -8,221 +8,433 @@
+    expect{ @lroads_client.put_async_relative_retry_no_status(@product).value! }.to raise_exception(MsRestAzure::AzureOperationError)
   end
 
   it 'should rise error if no provisioning state in payload provided on async PUT request' do
-    expect{ @client.lrosads.put_async_relative_retry_no_status_payload(@product).value! }.to raise_exception(MsRestAzure::AzureOperationError)
+    expect{ @lroads_client.put_async_relative_retry_no_status_payload(@product).value! }.to raise_exception(MsRestAzure::AzureOperationError)
   end
 
   it 'should rise error on invalid JSON responce on initial request' do
-    expect{ @client.lrosads.put200invalid_json(@product).value! }.to raise_exception(MsRest::DeserializationError)
+    expect{ @lroads_client.put200invalid_json(@product).value! }.to raise_exception(MsRest::DeserializationError)
   end
 
   it 'should rise error on invalid endpoint received in initial PUT request' do
-    expect{ @client.lrosads.put_async_relative_retry_invalid_header(@product).value! }.to raise_exception(MsRestAzure::AzureOperationError)
+    expect{ @lroads_client.put_async_relative_retry_invalid_header(@product).value! }.to raise_exception(MsRestAzure::AzureOperationError)
   end
 
   it 'should rise error on invalid JSON responce in status polling request during PUT operation' do
-    expect{ @client.lrosads.put_async_relative_retry_invalid_json_polling(@product).value! }.to raise_exception(MsRest::DeserializationError)
+    expect{ @lroads_client.put_async_relative_retry_invalid_json_polling(@product).value! }.to raise_exception(MsRest::DeserializationError)
   end
 
   it 'should rise error on invalid Location and Retry-After headers during DELETE operation' do
-    expect{ @client.lrosads.delete202retry_invalid_header().value! }.to raise_exception(MsRestAzure::AzureOperationError)
+    expect{ @lroads_client.delete202retry_invalid_header().value! }.to raise_exception(MsRestAzure::AzureOperationError)
   end
 
   it 'should rise error on invalid endpoint received in initial DELETE request' do
-    expect{ @client.lrosads.delete_async_relative_retry_invalid_header().value! }.to raise_exception(MsRestAzure::AzureOperationError)
+    expect{ @lroads_client.delete_async_relative_retry_invalid_header().value! }.to raise_exception(MsRestAzure::AzureOperationError)
   end
 
-  # TODO: Fix flakey test
-  #it 'should rise error on invalid JSON responce in status polling request during DELETE operation' do
-  #  expect{ @client.lrosads.delete_async_relative_retry_invalid_json_polling_async().value! }.to raise_exception(MsRest::DeserializationError)
-  #end
+  it 'should rise error on invalid JSON responce in status polling request during DELETE operation' do
+   expect{ @lroads_client.delete_async_relative_retry_invalid_json_polling().value! }.to raise_exception(MsRest::DeserializationError)
+  end
+
+  it 'should serve async PUT operation failed' do

```

Comment:
yes exactly same. removed it. thanks for catching 

---
In reply to: [69025710](https://github.com/Azure/autorest/pull/1188#discussion_r69025710) [](ancestors = 69025710)